src/test/java/mate/academy/springbookapp/controller/book/BookControllerTest.java

```diff
@@ -0,0 +1,264 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import javax.sql.DataSource;
+import lombok.SneakyThrows;
+import mate.academy.springbookapp.dto.book.BookDto;
+import mate.academy.springbookapp.dto.book.BookDtoWithoutCategoryIds;
+import mate.academy.springbookapp.dto.book.BookSearchParametersDto;
+import mate.academy.springbookapp.dto.book.CreateBookRequestDto;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerTest {
+    protected static MockMvc mockMvc;
+    private static final String CONTROLLER_ENDPOINT = "/api/books";
+    private static final String DB_PATH_ADD_THREE_BOOKS_WITH_CATEGORIES
+            = "database/book/add-three-books-with-categories.sql";
+    private static final String DB_PATH_REMOVE_ALL_BOOKS_WITH_CATEGORIES
+            = "database/book/remove-all-books-with-categories.sql";
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        init(dataSource, DB_PATH_ADD_THREE_BOOKS_WITH_CATEGORIES);
+    }
+
+    @AfterAll
+    static void afterAll(@Autowired DataSource dataSource) {
+        init(dataSource, DB_PATH_REMOVE_ALL_BOOKS_WITH_CATEGORIES);
```

Comment:
What I meant was to create methods:
- `init()` method for data initialization script called from `beforeAll`
- `teardown()` for data removal called from `afterAll`.
But I like your simplification better - I only see one issue: `init` name may be misleading since it's not initializing anything on `afterAll`. So how about `executeSqlScript()` instead?

Reply:
Thanks. It should be ok now :)